### DISCUSSION TITLE
Pages don't show last revision (GraphQL error: Cannot return null for non-nullable field ResponseStatus.slug.)

### DISCUSSION THREAD

**Discussion by missytake** [requarks/wiki#3340] [Error / Bug Report]
**Describe the bug**
If I edit and save a page, it throws the following error in the webUI: `GraphQL error: Cannot return null for non-nullable field ResponseStatus.slug.`. It aborts saving, I have to close the edit view manually. The page looks the same as before I tried to change it, but if I look at the page history, my changes *do* appear. So it just doesn't seem to show the last revision.

Also, sometimes if I try to click on "Save" during editing, it alerts me of a conflict, where I can choose between the local and the remote version, even if there are no differences between them. It says the local version is editable, while the remote version is write-protected. I'm not sure whether this is expected or not.

**To Reproduce**
1. Go to a page, click on edit
2. Make some changes
3. Click on "Save"
4. See error: `GraphQL error: Cannot return null for non-nullable field ResponseStatus.slug.`
5. Click on "Close", confirm that you want to discard changes
6. See that the page is unchanged.
7. Click on the revision history
8. See that all unsuccessful "Save" attempts are indeed written into the history.

**Expected behavior**
I expect that if I save a page, the changes are displayed, and that a page always shows the last revision.

**Screenshots**
If applicable, add screenshots to help explain your problem:

![Screenshot_2021-03-10_08-59-19](https://user-images.githubusercontent.com/57442838/110596080-28300f00-817f-11eb-8e21-4e4c1a656421.png)

![Screenshot_2021-03-10_08-59-31](https://user-images.githubusercontent.com/57442838/110596089-2b2aff80-817f-11eb-82e2-6209c491afff.png)

**Host Info (please complete the following information):**
 - OS: openBSD 6.8
 - Wiki.js version: 2.5.170
 - Database engine: sqlite

**Additional context**
- Browser: Firefox 86.0, reproducible in a private window
- Client OS: Xubuntu 20.04

**Comment by missytake** [comment]
If I retry the faulty request with curl (tokens censored):

```
curl 'https://wiki.heizhaus.org/graphql' \
            -H 'User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:86.0) Gecko/20100101 Firefox/86.0' \
            -H 'Accept: */*' \
            -H 'Accept-Language: en-US,en;q=0.5' \
            --compressed \
            -H 'Referer: https://wiki.heizhaus.org/e/de/home' \
            -H 'content-type: application/json' \
            -H 'Authorization: Bearer loremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtoken' \
            -H 'Origin: https://wiki.heizhaus.org' \
            -H 'DNT: 1' \
            -H 'Connection: keep-alive' \
            -H 'Cookie: jwt=loremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtokenloremipsumtoken' \
            --data-raw $'[{
                            "operationName":null,
                            "variables":{
                              "id":3,
                              "content":"# Das Heizhaus-Wiki\\n\\nAuf dieser Webseite findet ihr alle Informationen rund ums Heizhaus. *\\\\[Sollten hier nicht noch zwei generische Fragen stehen, um Leuten Lust darauf zu machen, weiterzulesen? Warum denkst du dir nicht welche aus? Immerhin ist das hier ein Wiki. **Du** kannst es editieren.]*\\n\\n## Wie erreiche ich die anderen Leute im Haus?\\n\\nAm einfachsten geht das \\xfcber [chat.heizhaus.org](https://chat.heizhaus.org), unseren Messenger.\\n\\nOder schreibt einfach eine Mail an [verteiler@heizhaus.org](mailto:verteiler@heizhaus.org).\\n\\n\\041[Das Heizhaus.](https://heizhaus.org/wp-content/uploads/2020/05/cropped-5F1A9D05-B428-4FAF-B60C-38BD28D274C5.jpg)",
                              "description":"wie finde ich mich in diesem wiki zurecht?",
                              "editor":"markdown",
                              "locale":"de",
                              "isPrivate":false,
                              "isPublished":true,
                              "path":"home",
                              "publishEndDate":"",
                              "publishStartDate":"",
                              "scriptCss":"",
                              "scriptJs":"",
                              "tags":[
                                
                              ],
                              "title":"\\xdcbersicht"
                            },
                            "extensions":{
                              
                            },
                            "query":"mutation ($id: Int\\041, $content: String, $description: String, $editor: String, $isPrivate: Boolean, $isPublished: Boolean, $locale: String, $path: String, $publishEndDate: Date, $publishStartDate: Date, $scriptCss: String, $scriptJs: String, $tags: [String], $title: String) {\\n  pages {\\n    update(id: $id, content: $content, description: $description, editor: $editor, isPrivate: $isPrivate, isPublished: $isPublished, locale: $locale, path: $path, publishEndDate: $publishEndDate, publishStartDate: $publishStartDate, scriptCss: $scriptCss, scriptJs: $scriptJs, tags: $tags, title: $title) {\\n      responseResult {\\n        succeeded\\n        errorCode\\n        slug\\n        message\\n        __typename\\n      }\\n      page {\\n        updatedAt\\n        __typename\\n      }\\n      __typename\\n    }\\n    __typename\\n  }\\n}\\n"
                          }]' 
```

I get the following response:

```
[
  {
    "errors": [
      {
        "message": "Cannot return null for non-nullable field ResponseStatus.slug.",
        "locations": [
          {
            "line": 7,
            "column": 9
          }
        ],
        "path": [
          "pages",
          "update",
          "responseResult",
          "slug"
        ],
        "extensions": {
          "code": "INTERNAL_SERVER_ERROR",
          "exception": {
            "stacktrace": [
              "Error: Cannot return null for non-nullable field ResponseStatus.slug.",
              "    at completeValue (/srv/wiki/node_modules/graphql/execution/execute.js:595:13)",
              "    at completeValueCatchingError (/srv/wiki/node_modules/graphql/execution/execute.js:530:19)",
              "    at resolveField (/srv/wiki/node_modules/graphql/execution/execute.js:461:10)",
              "    at executeFields (/srv/wiki/node_modules/graphql/execution/execute.js:297:18)",
              "    at collectAndExecuteSubfields (/srv/wiki/node_modules/graphql/execution/execute.js:748:10)",
              "    at completeObjectValue (/srv/wiki/node_modules/graphql/execution/execute.js:738:10)",
              "    at completeValue (/srv/wiki/node_modules/graphql/execution/execute.js:626:12)",
              "    at completeValue (/srv/wiki/node_modules/graphql/execution/execute.js:592:21)",
              "    at completeValueCatchingError (/srv/wiki/node_modules/graphql/execution/execute.js:530:19)",
              "    at resolveField (/srv/wiki/node_modules/graphql/execution/execute.js:461:10)",
              "    at executeFields (/srv/wiki/node_modules/graphql/execution/execute.js:297:18)",
              "    at collectAndExecuteSubfields (/srv/wiki/node_modules/graphql/execution/execute.js:748:10)",
              "    at completeObjectValue (/srv/wiki/node_modules/graphql/execution/execute.js:738:10)",
              "    at completeValue (/srv/wiki/node_modules/graphql/execution/execute.js:626:12)",
              "    at /srv/wiki/node_modules/graphql/execution/execute.js:527:16",
              "    at runMicrotasks (<anonymous>)",
              "    at processTicksAndRejections (internal/process/task_queues.js:97:5)",
              "    at async Promise.all (index 0)"
            ]
          }
        }
      }
    ],
    "data": {
      "pages": {
        "update": null,
        "__typename": "PageMutation"
      }
    }
  }
]
```

**Comment by missytake** [comment]
In the end, a reboot fixed the problem. So if anyone runs into the same bug, try rebooting the machine.

**Reply by missytake** [reply]
Unfortunately, this happens on such a regular basis that we can't just reboot the server every time it happens. We have to consider switching to another software.

**Reply by danmanners** [reply]
@missytake, I'm hitting this issue as well running the official containerized version running version 2.5.201. I'm going to try LinuxServer's, but wondering if you guys are still seeing this.

**Comment by danmanners** [comment]
If I attempt to Save or re-render **any** page, I encounter this GraphQL error as well.

![image](https://user-images.githubusercontent.com/17593824/126886816-dca67a2c-aabe-4441-99a6-a797cc5e07b0.png)

![image](https://user-images.githubusercontent.com/17593824/126886820-0e9efe40-4942-4714-8d5d-5e3f5ef26582.png)

I really want to use WikiJS, but literally can't currently.

All of my Kubernetes (K3s, v1.21.0) manifests for _how_ I've deployed it [can be found here](https://github.com/danmanners/homelab-k3s-cluster/tree/main/manifests/workloads/wikijs). I'm happy to provide whatever logs possible, but I'm finding very little so far.

Also, I'm connected to a PostgreSQL 13 Server that was spun up _only_ for this service. Don't know if there are any relevant logs I can provide from there.

**Comment by jrndberg** [comment]
I got the same error after trying to set this up. I setup LDAP and finally got that to work, filtered the OU and one user worked, added another user to the OU and then this started happening on all LDAP log in attempts. This was a deal breaker for me. Especially how easy and well documented LDAP authentication is these days, cool project but I might have to look elsewhere.

Also a reboot did not fix the issue for me. It's almost like it somehow killed it's own DB, as all queries I manually run results in the answer you got from the DB
